torch/_prims/__init__.py

```diff
@@ -2541,6 +2542,18 @@ def _scalar_tensor_aten(
     doc=_scalar_tensor_doc,
 )
 
+_int_tensor_doc = """
+    Wraps a SymInt or a list of SymInts into a Tensor with the specified dtype and device.
+"""
+
+int_tensor = _make_prim(
+    # TODO: handle scalar values
+    schema="int_tensor(SymInt[] data, *, ScalarType? dtype=None, Device? device=None, bool pin_memory=False, bool requires_grad=False) -> Tensor",
```

Comment:
It will be best if the prim takes shape in; otherwise you will have to expose unsafe_view to review the tensor to the nd shape after